lib/tag.js

```diff
@@ -60,6 +60,27 @@ var tag = function tag(tagName, attrsMap, content, contentIsEscaped) {
     return '<' + safeTagName + attrsHTML + (isSelfClosing(safeTagName) ? ' />' : '>' + safeContent + '</' + safeTagName + '>');
 };
 
+// expose
 tag.attrs = attrs;
 
+// --------------------
+
+var dataRegExp = /^data-[a-z]+([-][a-z]+)*$/;
+var ariaRegExp = /^aria-[a-z]+$/;
+var legalAttrs = ['autocomplete', 'autocorrect', 'autofocus', 'autosuggest', 'checked', 'dirname', 'disabled', 'tabindex', 'list', 'max', 'maxlength', 'min', 'novalidate', 'pattern', 'placeholder', 'readonly', 'required', 'size', 'step'];
+var ignoreAttrs = ['id', 'name', 'class', 'classes', 'type', 'value', 'multiple'];
+var getUserAttrs = function (opt) {
+    return reduce(opt, function (attributes, option, k) {
+        if ((ignoreAttrs.indexOf(k) === -1 && legalAttrs.indexOf(k) > -1) || dataRegExp.test(k) || ariaRegExp.test(k)) {
+            attributes[k] = option;
+        }
+        return attributes;
+    }, {});
+};
+
+// expose
+tag.getUserAttrs = getUserAttrs;
```

Comment:
I'm afraid it still needs all that information (that you're able to intuit) codified in explicit tests. Assume someone who has no idea what "getUserAttrs" means, wants to learn what the function does *only* by looking at its tests, including all edge cases.